app/jobs/in_person/enrollments_ready_for_status_check/enrollment_pipeline.rb

```diff
@@ -0,0 +1,144 @@
+      }
+
+      # Unwrap SQS message to get SNS message
+      begin
+        sns_message = JSON.parse(sqs_message.body, { symbolize_names: true })
+      rescue JSON::JSONError => err
+        report_error(err, **error_extra)
+        return false
+      end
+
+      unless sns_message.is_a?(Hash) && sns_message.key?(:MessageId) && sns_message.key?(:Message)
+        report_error('SQS message body is not valid SNS payload', **error_extra)
+        return false
+      end
+
+      error_extra[:sns_message_id] = sns_message[:MessageId]
+
+      # Unwrap SNS message to get SES message
+      begin
+        ses_message = JSON.parse(sns_message[:Message], { symbolize_names: true })
+      rescue JSON::JSONError => err
+        report_error(err, **error_extra)
+        return false
+      end
+
+      unless ses_message.is_a?(Hash) && ses_message.key?(:content) && ses_message.key?(:mail)
+        report_error('SNS "Message" field is not a valid SES payload', **error_extra)
+        return false
+      end
+
+      # Add information to help with debugging
+      error_extra[:ses_aws_message_id] = ses_message.dig(:mail, :messageId)
+      error_extra[:ses_mail_timestamp] = ses_message.dig(:mail, :timestamp)
+      error_extra[:ses_mail_source] = ses_message.dig(:mail, :source)
+
+      # https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.1
+      error_extra[:ses_rfc_origination_date] = ses_message.
+        dig(:mail, :commonHeaders, :date)&.tap do |date|
+          DateTime.parse(date).to_s
+        end
+      # https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.4
+      error_extra[:ses_rfc_message_id] = ses_message.dig(:mail, :commonHeaders, :messageId)
+
+      # Parse email from content of SES message
+      begin
+        mail = Mail.read_from_string(ses_message[:content])
+        # Depending on how the email is created, we may need to read different
+        # parts of the message
+        if mail.multipart?
+          text_body = mail.text_part&.decoded
+        else
+          text_body = mail.body.decoded
+        end
+      rescue StandardError
+        report_error(err, **error_extra)
+        return false
+      end
+
+      unless text_body.respond_to?(:to_s) && text_body.to_s.present?
+        report_error('Failure occurred when attempting to get email body', **error_extra)
+        return false
+      end
+
+      # Extract enrollment code from email body
+      enrollment_code = email_body_pattern.match(text_body.to_s)&.[](:enrollment_code)
+
+      unless enrollment_code.is_a?(String)
+        report_error(
+          'Failed to extract enrollment code using regex, check email body format and regex',
+          **error_extra,
+        )
+        return false
+      end
+
+      error_extra[:enrollment_code] = enrollment_code
+
+      # Look up existing enrollment
+      id, user_id, ready_for_status_check = InPersonEnrollment.
+        where(enrollment_code:, status: :pending).
```

Comment:
here we filter on status, later we update ready_for_status_check, are they the same thing?

Reply:
No, they currently track different things. `status` tracks the overall progress of the enrollment creation, checking, and success/failure. `ready_for_status_check` only covers whether we received a message saying that the enrollment is ready to have its status checked (which is not tracked by `status`).

The check with USPS may result in a `status` update if USPS indicates that the enrollment is not present, has expired, or that the user has visited the post office already.

Reply:
More context here - we are currently checking enrollments that have a `pending` status. When this and related stories are complete, we will be able to deprioritize (but still eventually check) enrollments that have `ready_for_status_check` remain as `false`.